### PR TITLE
[Fix] Fix multi-view inference

### PR DESCRIPTION
**Pull request by cir7**
## Modification
[fix] fix multiple crop inference, now support save mutli scores of crops

### CODE REVIEW

**Review by Dai-Wenxun — APPROVED** [review]

File: mmaction/models/recognizers/base.py [hunk 1]
```diff
@@ -237,20 +237,22 @@ def forward(self,
             raise RuntimeError(f'Invalid mode "{mode}". '
                                'Only supports loss, predict and tensor mode')
 
-    def convert_to_datasample(self, inputs: SampleList,
-                              data_samples: InstanceList) -> SampleList:
-        """Convert predictions to ``ActionDataSample``.
+    def wrap_in_datasample(self, data_samples: SampleList,
```
Comment (ly015) [inline mmaction/models/recognizers/base.py:240]: I think this part can be moved into `predict()`. No post-processing or formatting is involved, so I guess there is no need to define a method.

File: mmaction/models/heads/base.py [hunk 2]
```diff
@@ -225,13 +225,12 @@ class score. Only called in test mode.
...
                              f'["score", "prob", None]')
 
-        if self.average_clips is None:
-            return cls_scores
-
         batch_size = cls_scores.shape[0]
         cls_scores = cls_scores.view(batch_size // num_segs, num_segs, -1)
 
-        if self.average_clips == 'prob':
+        if self.average_clip is None:
```
Comment (ly015) [inline mmaction/models/heads/base.py:231]: `self.average_clips`?

Comment (ly015) [reply mmaction/models/heads/base.py:231]: BTW, is this part covered by ut?

File: mmaction/models/recognizers/base.py (hunk 1 above)
Comment (cir7) [reply mmaction/models/recognizers/base.py:240]: that's good

File: mmaction/datasets/transforms/formatting.py [hunk 3]
```diff
@@ -169,12 +169,18 @@ def __repr__(self):
...
 
-    Required keys are ``imgs``, ``num_clips`` and ``clip_len``,
-    added or modified keys are ``imgs`` and ``input_shape``.
+    Required keys:
+        - imgs
+        - num_clips
+        - clip_len
+
+    Modified Keys:
+        - img
```
Comment (Dai-Wenxun) [inline mmaction/datasets/transforms/formatting.py:178]:
```suggestion
        - imgs
```

**Pull request merged**